doc/model.rst

```diff
@@ -0,0 +1,719 @@
+- first_device (int = 0): The first GPU device to use on the host. This parameter only applies to GPU devices and will be ignored if device is specified as CPU.
+- batch_size (int = 0): batch size for execution, defaults to 0.
+- min_batch_size (int = 0): minimum batch size for model execution, defaults to 0.
+- min_batch_timeout (int = 0): time to wait for minimum batch size, defaults to 0.
+- tag (str = ""): additional tag for model information, defaults to “”.
+- inputs (t.Optional[t.List[str]] = None): model inputs (TF only), defaults to None.
+- outputs (t.Optional[t.List[str]] = None): model outputs (TF only), defaults to None.
+
+These arguments provide details to add and configure
+ML models within the model simulation.
+
+Example: Loading an In-Memory ML Model to the Model
+---------------------------------------------------
+This example demonstrates how to instruct SmartSim to load
+an in-memory ML model into the database at model runtime.
+
+**Python Script: Creating a Keras CNN for Model Purposes**
+
+To create an in-memory ML model, define the Model within the Python driver script.
+For the purpose of the example, we define a Keras CNN within the experiment.
+
+.. code-block:: python
+
+    def create_tf_cnn():
+        """Create a Keras CNN for model purposes
+
+        """
+        from smartsim.ml.tf import serialize_model
+        n = Net()
+        input_shape = (3,3,1)
+        inputs = Input(input_shape)
+        outputs = n(inputs)
+        model = keras.Model(inputs=inputs, outputs=outputs, name=n.name)
+
+        return serialize_model(model)
+
+    # Get and save TF model
+    model, inputs, outputs = create_tf_cnn()
+
+**SmartSim Model Integration:**
+
+Assuming an initialized ``Model`` named `smartsim_model` exists, we add a TensorFlow model using
+the ``Model.add_ml_model()`` function:
+
+.. code-block:: python
+
+    smartsim_model.add_ml_model(name="cnn", backend="TF", model=model, device="GPU", devices_per_node=2, first_device=0, inputs=inputs, outputs=outputs)
+
+In the above ``smartsim_model.add_ml_model()`` code snippet, we offer the following arguments:
+
+-  name ("cnn"): A key to uniquely identify the model within the database.
+-  backend ("TF): Indicating that the model is a TensorFlow model.
+-  model (model): The in-memory representation of the TensorFlow model.
+-  device ("GPU"): Specifying the device for ML model execution.
+-  devices_per_node (2): Use two GPUs per node.
+-  first_device (0): Start with 0 index GPU.
+-  inputs (inputs): The names of the model inputs.
+-  outputs (outputs): The names of the model outputs.
+
+.. warning::
+    Calling `exp.start(model)` prior to instantiation of an orchestrator will result in
+    a failed attempt to load the ML model to a non-existent database.
+
+--------------------
+Tensorflow functions
```

Comment:
I think there is a mixup here.  There are only TorchScript functions supported by SmartSIm.  I think this section is referring to the ability of SmartSim to directly take a function that is in Python file and pass the function directly to ``add_function`` and executed as a TorchScript in the Orchestrator.  This section and the following section need to be combined and reworked.